front/API.ts

```diff
@@ -269,4 +269,48 @@ export default class API {
 			["https://media.discordapp.net/attachments/717080637038788731/1067469563073142804/vivaldi_split_8.png", 1868, 400],
 		];
 	}
+
+	/**
+	 * 
+	 * @param dataKey
+	 * @param value
+	 * @description update user's account credentials. Either email or password, you choose via the datakey param
+	 * @returns new user's credentials
+	 */
+
+	static async updateUserCredentials(dataKey: 'password' | 'email', oldValue: string, newValue: string): Promise<string> {
+		return new Promise<string>((resolve, reject) => {
+			setTimeout(() => {
+				try {
+					if (dataKey == 'email') {
+						API.fetch({route: '/auth/me'}).then((res) => {
+							if (res.email === oldValue) {
+								API.fetch({
+									route: '/auth/me',
+									method: "PATCH",
+									body: {'email': newValue}}).finally(() => {
+										return resolve('email successfully changed');
+									});
+							} else return reject('Wrong email');
+						});
+					} else if (dataKey == 'password') {
+						API.fetch({route: '/auth/me'}).then((res) => {
+							if (res.password === oldValue) {
```

Comment:
The /auth/me does not return the password. This would mean that the pw is stored in clear in the db. Moreover, the body of the request updates the email, not the password

Reply:
is that okay if i just return an error for the password update ? I said during the Delivery that it was not functional.

Reply:
Yes I guess it's fine then

Reply:
feature is working, ready to merge